server/routes/opensearch_routes_service.ts

```diff
@@ -429,4 +526,59 @@ export class OpenSearchRoutesService {
       return generateCustomError(res, err);
     }
   };
+
+  getIngestPipeline = async (
+    context: RequestHandlerContext,
+    req: OpenSearchDashboardsRequest,
+    res: OpenSearchDashboardsResponseFactory
+  ): Promise<IOpenSearchDashboardsResponse<any>> => {
+    const { pipeline_id } = req.params as { pipeline_id: string };
+    const { data_source_id = '' } = req.params as { data_source_id?: string };
+
+    try {
+      const callWithRequest = getClientBasedOnDataSource(
+        context,
+        this.dataSourceEnabled,
+        req,
+        data_source_id,
+        this.client
+      );
+
+      const response = await callWithRequest('ingest.getPipeline', {
+        id: pipeline_id,
+      });
+
+      return res.ok({ body: response });
+    } catch (err: any) {
+      return generateCustomError(res, err);
+    }
+  };
+
+  getSearchPipeline = async (
+    context: RequestHandlerContext,
+    req: OpenSearchDashboardsRequest,
+    res: OpenSearchDashboardsResponseFactory
+  ): Promise<IOpenSearchDashboardsResponse<any>> => {
+    const { pipeline_id } = req.params as { pipeline_id: string };
+    const { data_source_id = '' } = req.params as { data_source_id?: string };
+
+    try {
+      const callWithRequest = getClientBasedOnDataSource(
+        context,
+        this.dataSourceEnabled,
+        req,
+        data_source_id,
+        this.client
+      );
+
+      const response = await callWithRequest('transport.request', {
+        method: 'GET',
+        path: SEARCH_PIPELINE_ROUTE + '/' + pipeline_id,
+      });
```

Comment:
Is there not a node js client path for getting search pipelines, like for the other core apis? (e.g., `ingest.getPipeline`)?

If not, I'd suggest to add this path in `server/cluster` somewhere. Note we have the defined paths to execute different plugin APIs for Flow Framework and ML commons there. Maybe, would need a new "core" or "search pipelines" plugin to logically persist them there.

Reply:
@ohltyler, I initially tried by adding in server/cluster. But after referring AD frontend plugin [here](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/d8f738076e265dc858800573529469ca6d13ce19/server/routes/opensearch.ts#L176) ,  using `transport.request `felt right. Instead of adding other file.

Reply:
Ok, my concern is that this is objectively weaker and we don't have as much type safety or guardrails around the API than if we explicitly define, AD should not have done it that way. Not a blocker though. Also, please double check the JS client as it seems this should be available out of the box. Not sure if search pipelines APIs got missed?

Reply:
Will followup on this

Reply:
IMHO it's less effort to make the change then follow up with tracking issues etc. 😄  But please do one of them before resolving the conversation so it is tracked accordingly.

Reply:
This is added [here](https://github.com/opensearch-project/dashboards-flow-framework/pull/462/commits/b5c9818164e9034b4bd2ac95378659d4a6a8ae85)